single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/builders/NetworkGraphBuilder.java

```diff
@@ -346,7 +371,17 @@ public void visitShuntCompensator(ShuntCompensator sc) {
 
         @Override
         public void visitDanglingLine(DanglingLine dl) {
-            addTerminalNode(NodeFactory.createDanglingLine(graph, dl.getId(), dl.getNameOrId()), dl.getTerminal());
+            if (!dl.isPaired()) {
+                addTerminalNode(NodeFactory.createDanglingLine(graph, dl.getId(), dl.getNameOrId()), dl.getTerminal());
+            } else {
+                dl.getTieLine().ifPresent(tieLine -> visitTieLine(tieLine, dl, graph));
+            }
+        }
+
+        private void visitTieLine(TieLine tieLine, DanglingLine dl, Graph graph) {
```

Comment:
To be checked with @flo-dup and @miovd but it seems that we have nothing that prevents two dangling lines, connected to the same voltage level and paired together. It is a "bug" of usage and I think that in the case, the user wants to see the wierd tie line. I wonder if we are supporting that here.

Reply:
It seems that you are supporting that looking at the last picture of your PR!